unittests/test_fixtures.py

```diff
@@ -319,9 +324,26 @@ def _fixture_wrapper(**kwargs):
     yield _fixture_wrapper
 
 
+@pytest.fixture
+def simple_test():
+    def _make_test(vs, ve):
+        class _X(rfm.RegressionTest):
+            valid_systems = vs
+            valid_prog_environs = ve
+
+        return _X()
+
+    return _make_test
+
+
 def test_fixture_registry_all(ctx_sys, simple_fixture):
     '''Test with all valid partition and environments available.'''
 
+    class MyTest(rfm.RegressionTest):
```

Comment:
You can use the `simple_test` fixture here

Reply:
In the context of these unit tests we need the test just for its `valid_systems`  and `valid_prog_environs`. See it as a wrapper in this case and a normal test. The registry's `add()` method takes the fixture to be registered as a separate argument and not from the `parent_test` argument.